projects/packages/sync/src/sync-queue/class-queue-storage-table.php

```diff
@@ -456,4 +476,190 @@ public function delete_items_by_ids( $ids ) {
+		}
+
+		if ( ! $custom_table_instance->is_custom_table_healthy() ) {
+
+			// TODO: clean up the table.
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Migrates the existing Sync events from the options table to the Custom table
+	 *
+	 * @return void
+	 */
+	public static function migrate_from_options_table_to_custom_table() {
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching
+		$count_result = $wpdb->get_row(
+			"
+				SELECT
+					COUNT(*) as item_count
+				FROM
+				    {$wpdb->options}
+				WHERE
+				    option_name LIKE 'jpsq_%'
+			"
+		);
+
+		$item_count = $count_result->item_count;
+
+		$limit  = 100;
+		$offset = 0;
+
+		do {
+			// get all the records from the options table
+			$query = "
+				SELECT
+					option_name as event_id,
+					option_value as event_payload
+				FROM
+				    {$wpdb->options}
+				WHERE
+				    option_name LIKE 'jpsq_%'
+				ORDER BY
+				    option_name ASC
+				LIMIT $offset, $limit
+			";
+
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching,WordPress.DB.PreparedSQL.NotPrepared
+			$rows = $wpdb->get_results( $query );
+
+			$insert_rows = array();
+
+			foreach ( $rows as $event ) {
+				$event_id = $event->event_id;
+
+				// Parse the event
+				if (
+					preg_match(
+						'!jpsq_(?P<queue_id>[^-]+)-(?P<timestamp>[^-]+)-.+!',
+						$event_id,
+						$events_matches
+					)
+				) {
+					$queue_id  = $events_matches['queue_id'];
+					$timestamp = $events_matches['timestamp'];
+
+					$insert_rows[] = $wpdb->prepare(
+						'(%s, %s, %s, %s)',
+						array(
+							$queue_id,
+							$event_id,
+							$event->event_payload,
+							(int) $timestamp,
+						)
+					);
+				}
+			}
+
+			// Instantiate table storage, so we can get the table name. Queue ID is just a placeholder here.
+			$queue_table_storage = new Queue_Storage_Table( 'test_queue' );
+
+			if ( ! empty( $insert_rows ) ) {
+				$insert_query = 'INSERT INTO ' . $queue_table_storage->table_name . ' (queue_id, event_id, event_payload, timestamp) VALUES ';
+
+				$insert_query .= implode( ',', $insert_rows );
+
+				// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching,WordPress.DB.PreparedSQL.NotPrepared
+				$wpdb->query( $insert_query );
+			}
+
+			$offset += $limit;
+		} while ( $offset < $item_count );
+
+		// Clear out the options queue
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching,WordPress.DB.PreparedSQL.NotPrepared
+		$wpdb->query(
+			$wpdb->prepare(
+				"DELETE FROM $wpdb->options WHERE option_name LIKE %s",
+				'jpsq_%-%'
+			)
+		);
+	}
+
+	/**
+	 * Migrates the existing Sync events from the Custom table to the Options table
+	 *
+	 * @return void
+	 */
+	public static function migrate_from_custom_table_to_options_table() {
+		global $wpdb;
```

Comment:
Do you think we should delete any leftover options first before populating them from the custom table?